rules/framework.bzl

```diff
@@ -340,20 +355,87 @@ def _apple_framework_packaging_impl(ctx):
             swift_common.create_module(name = swiftmodule_name, swift = swift_module),
         ]
 
-    # Eventually we need to remove any reference to objc provider
-    # and use CcInfo instead, see this issue for more details: https://github.com/bazelbuild/bazel/issues/10674
-    objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
+    direct_headers = []
+
+    if virtualize_frameworks:
+        import_vfsoverlays = []
+        for dep in ctx.attr.vfs:
+            if not VFSOverlayInfo in dep:
+                continue
+            import_vfsoverlays.append(dep[VFSOverlayInfo].vfs_info)
+
+        # Progated interface headers - this must encompass all of them
+        propagated_interface_headers = []
+
+        # We need to map all the deps here - for both swift headers and others
+        fw_dep_vfsoverlays = []
+        for dep in ctx.attr.transitive_deps + ctx.attr.deps:
+            if not FrameworkInfo in dep:
+                continue
+            framework_info = dep[FrameworkInfo]
+            fw_dep_vfsoverlays.extend(framework_info.vfsoverlay_infos)
+            propagated_interface_headers.append(framework_info.framework_headers)
+
+            # Collect generated headers. consider exposing all required generated
+            # headers in respective providers: -Swift, modulemap, -umbrella.h
+            if not CcInfo in dep:
+                continue
+            for h in dep[CcInfo].compilation_context.headers.to_list():
+                if h.is_source:
+                    continue
+                propagated_interface_headers.append(depset([h]))
+
+        vfs_ret = make_vfsoverlay(
+            ctx,
+            hdrs = header_out,
+            module_map = modulemap_out,
+            private_hdrs = private_header_out,
+            has_swift = len(swiftmodule_out) > 0,
+            merge_vfsoverlays = import_vfsoverlays + fw_dep_vfsoverlays,
+        )
+
+        framework_headers = depset(header_out + modulemap_out + private_header_out)
+
+        # Includes interface headers here ( handled in cc_info merge for no virutal )
+        _add_to_dict_if_present(compilation_context_fields, "headers", depset(
+            direct = header_out + private_header_out + modulemap_out,
+            transitive = propagated_interface_headers,
+        ))
+        framework_info = FrameworkInfo(
+            vfsoverlay_infos = [vfs_ret.vfs_info],
+            framework_headers = framework_headers,
+        )
+
+    else:
```

Comment:
i don't expect an else stmt here, I expect the previous if not virtualized_framework handles everything. 
so this is empty for not_virtualized framework correct?